app/models/course/forum/topic.rb

```diff
@@ -10,6 +10,60 @@ class Course::Forum::Topic < ActiveRecord::Base
   has_many :posts, class_name: Course::Discussion::Post.name, through: :topic
   belongs_to :forum, inverse_of: :topics
 
+  # @!attribute [r] post_count
+  #   The number of posts in this topic.
+  calculated :post_count, (lambda do
+    from('course_discussion_topics TOPIC INNER JOIN course_discussion_posts
+      POST On TOPIC.id = POST.topic_id').select { count('*') }.
+      where('TOPIC.actable_id = course_forum_topics.id AND TOPIC.actable_type = ?', name).to_sql
+  end)
+
+  # @!attribute [r] view_count
+  #   The number of views in this topic.
+  calculated :view_count, (lambda do
+    Course::Forum::Topic::View.where do
+      course_forum_topic_views.topic_id == course_forum_topics.id
+    end.select { count('*') }.to_sql
+  end)
+
+  # @!method self.with_latest_post
+  #   Augments all returned records with the latest post.
+  scope :with_latest_post, (lambda do
+    ids = Course::Discussion::Post.select { max(id) }.group { topic_id }
+    last_posts = Course::Discussion::Post.where(id: ids).includes(:creator)
+
+    all.tap do |result|
+      preloader = ActiveRecord::Associations::Preloader::ManualPreloader.new
+      preloader.preload(result, :posts, last_posts)
+    end
+  end)
+
+  def votes_count
+    0 # :TODO
+  end
+
+  # Return if a user has subscribed to this topic
+  #
+  # @param [User] user The user to check
+  # @return [Boolean] True if the user has subscribed this topic
+  def subscribed_by?(user)
+    subscriptions.where(user: user).any?
+  end
+
+  # Create a topic with an initial post whose title is the topic's title
+  #
+  # @param [Hash] post_params The parameters used to create the post
+  # @return [Boolean] True if the topic and post are created successfully
+  def create_with_post(post_params)
```

Comment:
Sorry my fault. You are right. But for now, the problems come from `Discussion::Topic`. If I want to use nested attributes, I need to build post using topic. But actually, I need to do like this: `@topic.topic.posts.build`. The middle topic is the `Discussion::Topic`. Problem is that `@topic.topic` will be nil before the @topic is saved successfully. @lowjoel